zulipterminal/ui_tools/boxes.py

```diff
@@ -78,6 +87,34 @@ def stream_box_view(self, caption: str='', title: str='') -> None:
         ]
         self.contents = write_box
 
+    def generic_autocomplete(self, text: str, state: int) -> Optional[str]:
+        if text.startswith('@'):
+            return self.autocomplete_mentions(text, state)
+        elif text.startswith('#'):
+            return self.autocomplete_streams(text, state)
+        else:
+            return text
+
+    def autocomplete_mentions(self, text: str, state: int) -> Optional[str]:
+        users_list = self.view.users
+        user_typeahead = ['@**%s**' % user['full_name']
```

Comment:
We also need to have an additional check here. If you have two users with the same name, the syntax we use is: `@**Name|userid**`. I wonder how we could differentiate them on the terminal though (maybe showing email?).

Reply:
Great! Thanks @aero31aero .